app/routes/calendar.py

```diff
@@ -30,69 +31,123 @@ def post(self, location_id):
+            # Czechy way of invoking the helper function
+            if check_for_overlap(dates, req_start_date, req_end_date):
+                return check_for_overlap(dates, req_start_date, req_end_date)
+
             # if the requested dates do not envelope or are enveloped by an existing date range, commit the selected dates to the database
             calendar = Calendar(**data)
             db.session.add(calendar)
             db.session.commit()
             return {"message": "Successfully scheduled!"}
 
-    @api.routes("/<int:id>")
-    @api.response(404, "Calendar Booking not found")
-    class CalendarById(Resource):
-        def get(self, id):
-            '''Get a Calendar date for the provided id'''
-            calendar_range = Calendar.query.get(int(id))
-            if calendar_range:
-                return {"calendar":calendar_range.to_dictionary()}
-            else:
-                return {"message": "Calendar Entry Not Found"}, 404
-        @api.expect(model)
-        def put(self, id):
-            '''Update calendar by calendar id using the data passed in'''
-            calendar_range = Calendar.query.get(int(id))
-            if calendar_range:
-                data = api.payload
-
-                calendar_range.start_date = data["start_date"]
-                calendar_range.end_date = data["end_date"]
-
-                db.session.commit()
-
-                return {"message": "Calendar Booking was successfully updated!"}
-            else:
-                return {"message": "Calendar Booking was not found"}, 404
-
-        def delete(self, id):
-            '''Delete Calendar Booking for the provided calendar id'''
-            calendar_range = Calendar.query.get(int(id))
-            if calendar_range:
-                db.session.delete(calendar_range)
-                db.session.commit()
-                return {"message": "Calendar Booking deleted successfully"}
-            else:
-                return {"message": "Calendar Booking not found, nothing deleted"}, 404
+@api.route("/<int:id>")
+@api.response(404, "Calendar Booking not found")
+@api.param('id', 'The calendar identifier')
+@api.param('location_id', 'The location identifier')
+class CalendarById(Resource):
+    def get(self, location_id, id):
+        '''Get a Calendar date for the provided id'''
+        calendar_range = Calendar.query.get(int(id))
+        if calendar_range:
+            return {"calendar":calendar_range.to_dictionary()}
+        else:
+            return {"message": "Calendar Entry Not Found"}, 404
+    @api.expect(model)
+    def put(self, location_id, id):
+        '''Update calendar by calendar id using the data passed in'''
+        # get the specifc calendar date to edit
+        calendar_range = Calendar.query.get(int(id))
+        # get all the dates for the object to update on
+        database_dates = Calendar.query.filter_by(location_id=location_id).all()
```

Comment:
what is the use case here? Aren't we already getting the calendar record on line 84?

Reply:
calendar_ranges is the individual calendar range that we want to update
database_dates are all of the dates for the given location that we want to check against to make sure that the update does not overlap with a pre-existing calendar database entry.